test/app/features/profile/domain/usecases/user_profile_test.dart

```diff
@@ -1,90 +1,115 @@
-      final actual = right(const ValidField());
-      when(repository.stealthMode(toggle: anyNamed('toggle')))
-          .thenAnswer((_) async => right(const ValidField()));
-      // act
-      final expected = await sut.stealthMode(toggle: false);
-      // assert
-      expect(actual, expected);
-      verify(repository.stealthMode(toggle: false));
-    });
+        // act
+        final expected = await sut.stealthMode(toggle: true);
+        // assert
+        expect(actual, expected);
+        verify(() => repository.stealthMode(toggle: true)).called(1);
+        verify(() => appStateUseCase.check()).called(1);
+      },
+    );
 
-    test('should enable anonymous mode', () async {
-      // arrange
-      final actual = right(const ValidField());
-      when(repository.anonymousMode(toggle: anyNamed('toggle')))
-          .thenAnswer((_) async => right(const ValidField()));
-      // act
-      final expected = await sut.anonymousMode(toggle: true);
-      // assert
-      expect(actual, expected);
-      verify(repository.anonymousMode(toggle: true));
-    });
+    test(
+      'disable stealth mode',
+      () async {
+        // arrange
+        final actual = right(const ValidField());
+        when(() => repository.stealthMode(toggle: any(named: 'toggle')))
+            .thenAnswer((_) async => right(const ValidField()));
+        // act
+        final expected = await sut.stealthMode(toggle: false);
+        // assert
+        expect(actual, expected);
+        verify(() => repository.stealthMode(toggle: false)).called(1);
+      },
+    );
 
-    test('should disable anonymous mode', () async {
-      // arrange
-      final actual = right(const ValidField());
-      when(repository.anonymousMode(toggle: anyNamed('toggle')))
-          .thenAnswer((_) async => right(const ValidField()));
+    test(
+      'should enable anonymous mode',
+      () async {
+        // arrange
+        final actual = right(const ValidField());
+        when(() => repository.anonymousMode(toggle: any(named: 'toggle')))
+            .thenAnswer((_) async => right(const ValidField()));
+        // act
+        final expected = await sut.anonymousMode(toggle: true);
+        // assert
+        expect(actual, expected);
+        verify(() => repository.anonymousMode(toggle: true)).called(1);
+      },
+    );
 
-      // act
-      final expected = await sut.anonymousMode(toggle: false);
-      // assert
-      expect(actual, expected);
-      verify(repository.anonymousMode(toggle: false));
-      verify(appStateUseCase.check());
-    });
+    test(
+      'should disable anonymous mode',
+      () async {
+        // arrange
+        final actual = right(const ValidField());
+        when(() => repository.anonymousMode(toggle: any(named: 'toggle')))
+            .thenAnswer((_) async => right(const ValidField()));
+        // act
+        final expected = await sut.anonymousMode(toggle: false);
+        // assert
+        expect(actual, expected);
```

Comment:
Sei que foi mantida a estrutura dos testes, porém poderia aproveitar e ajustar o nome das variáveis? Pois caso o teste falhe, a mensagem pode ficar um pouco confuso de se entender sendo que apareceria que o retorno de `anonymousMode` (que estaria errado) é o esperado.

```suggestion
        final expected = right(const ValidField());
        when(() => repository.anonymousMode(toggle: any(named: 'toggle')))
            .thenAnswer((_) async => right(const ValidField()));
        // act
        final actual = await sut.anonymousMode(toggle: false);
        // assert
        expect(actual, expected);
```